rs/bitcoin/ckbtc/minter/src/updates/update_balance.rs

```diff
@@ -283,60 +283,45 @@ pub async fn update_balance(
     Ok(utxo_statuses)
 }
 
-async fn kyt_check_utxo(
-    caller: Principal,
-    utxo: &Utxo,
-) -> Result<(String, UtxoCheckStatus, Principal), UpdateBalanceError> {
-    let kyt_principal = read_state(|s| {
-        s.kyt_principal
-            .expect("BUG: upgrade procedure must ensure that the KYT principal is set")
+async fn kyt_check_utxo(utxo: &Utxo) -> Result<UtxoCheckStatus, UpdateBalanceError> {
+    let new_kyt_principal = read_state(|s| {
+        s.new_kyt_principal
+            .expect("BUG: upgrade procedure must ensure that the new KYT principal is set")
             .get()
             .into()
     });
 
-    if let Some((uuid, status, api_key_owner)) = read_state(|s| s.checked_utxos.get(utxo).cloned())
-    {
-        return Ok((uuid, status, api_key_owner));
+    if let Some((_, status, _)) = read_state(|s| s.checked_utxos.get(utxo).cloned()) {
+        return Ok(status);
     }
 
-    match fetch_utxo_alerts(kyt_principal, caller, utxo)
+    match check_transaction(new_kyt_principal, utxo)
         .await
         .map_err(|call_err| {
             UpdateBalanceError::TemporarilyUnavailable(format!(
                 "Failed to call KYT canister: {}",
                 call_err
             ))
         })? {
-        Ok(response) => {
-            if !response.alerts.is_empty() {
-                log!(
-                    P0,
-                    "Discovered a tainted UTXO {} (external id {})",
-                    DisplayOutpoint(&utxo.outpoint),
-                    response.external_id
-                );
-                Ok((
-                    response.external_id,
-                    UtxoCheckStatus::Tainted,
-                    response.provider,
-                ))
-            } else {
-                Ok((
-                    response.external_id,
-                    UtxoCheckStatus::Clean,
-                    response.provider,
-                ))
-            }
+        CheckTransactionResponse::Failed(addresses) => {
+            log!(
+                P0,
+                "Discovered a tainted UTXO {} (due to input addresses {})",
```

Comment:
nit: should we also log the `UpdateBalanceArgs` (principal and subaccount) that triggered this? I have the impression this would be useful for debuggability .

Reply:
The changes should be in the code. I think github either redirects users to the old version if you follow the linked code above,  or it just won't show related discussion thread if you go to File Changed tag to view the code & comments.